packages/republik/lib/MembershipStats/geo.js

```diff
@@ -0,0 +1,139 @@
+const debug = require('debug')('republik:lib:MembershipStats:geo')
+const moment = require('moment')
+
+const {
+  cache: { create },
+} = require('@orbiting/backend-modules-utils')
+
+const QUERY_CACHE_TTL_SECONDS = 60 * 60 * 24 * 8 // A week and a day
+
+const postalCodeParsers = {
+  CH: (code) => parseInt(code.replace(/^CH[\s-]*/i, '')).toString(),
+  DE: (code) => code.replace(/^D[\s-]*/i, '').split(' ')[0],
+  AT: (code) => code.replace(/^A[\s-]*/i, '').split(' ')[0],
+  BE: (code) => code.replace(/^B[\s-]*/i, '').split(' ')[0],
+  DK: (code) => code.replace(/^DK[\s-]*/i, '').split(' ')[0],
+  IT: (code) => code.replace(/^I[\s-]*/i, '').split(' ')[0],
+  NL: (code) => parseInt(code.replace(/^NL[\s-]*/i, '')).toString(),
+}
+
+const createCache = (context) =>
+  create(
+    {
+      namespace: 'republik',
+      prefix: 'MembershipStats:geo',
+      key: 'any',
+      ttl: QUERY_CACHE_TTL_SECONDS,
+    },
+    context,
+  )
+
+const populate = async (context, resultFn) => {
+  debug('populate')
+
+  const { pgdb } = context
+
+  const memberships = await pgdb.query(`
+    SELECT
+      m.id,
+      MIN(mp."beginDate") "minBeginDate",
+      MAX(mp."endDate") "maxEndDate",
+      a."country",
+      a."postalCode"
+    FROM "memberships" m
+    JOIN "membershipPeriods" mp
+      ON mp."membershipId" = m.id
+    JOIN "users" u
+      ON u.id = m."userId"
+    LEFT JOIN "addresses" a
+      ON a.id = u."addressId"
+    WHERE m."userId" != 'f0512927-7e03-4ecc-b14f-601386a2a249' -- Jefferson
+    GROUP BY m.id, a.id
+  `)
+
+  const countries = await pgdb.public.statisticsGeoCountry.findAll()
+
+  const geo = {}
+
+  for (const membership of memberships) {
+    const country = countries.find((country) => {
+      return (
+        country.name === membership.country ||
+        country.searchNames.indexOf(membership.country?.toLowerCase()) > -1
+      )
+    })
```

Comment:
```suggestion
    const country = !!membership.country.trim() && countries.find((country) => {
      return (
        country.name === membership.country ||
        country.searchNames.includes(membership.country.toLowerCase().trim())
      )
    })
```

Could be a slight speed improvement: Checking first if country is truthy before digging in countries array.

Do you think `.includes` is faster than `.indexOf`? I find it more readable.

Also, trimming might increase likely-hood of finding a matching country.